src/main/webapp/WEB-INF/view/activityfeed.jsp

```diff
@@ -16,15 +43,67 @@
       <a href="/login">Login</a>
     <% } %>
     <a href="/about.jsp">About</a>
+    <a href="/activityfeed">Activity Feed</a>
   </nav>
 
   <div id="container">
+    <div
+      style="width:75%; margin-left:auto; margin-right:auto; margin-top: 50px;">
     <h1>Activity Feed</h1>
 
     <% if(request.getAttribute("error") != null){ %>
         <h2 style="color:red"><%= request.getAttribute("error") %></h2>
     <% } %>
- 	<p> This is the Activity Feed </p>
+ 	<p>	Here's everything that's happened on the site so far!</p>
+      
+    <div id="activityfeed">  
+      <ul>
+    <%
+ 	ArrayList<Activity> events = new ArrayList<Activity>();
+ 	events.addAll(PersistentStorageAgent.getInstance().loadUsers());
```

Comment:
Rather than getting a new instance each time, it would be more efficient to create a variable for PersistentStorageAgent.getInstance() and reuse it on each of these lines.

Reply:
Done.